JS/Lesson34/gulcin/recipe-app/src/modules/recipes/NewRecipe/index.jsx

```diff
@@ -0,0 +1,179 @@
+import { useForm } from 'react-hook-form';
+import { v4 as createId } from 'uuid';
+import './styles.css';
+
+
+export const NewRecipe = () => {
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+    } = useForm();
+
+
+    const ingredientIndexes = Array.from({ length: 20 }, (_, i) => i + 1);
+
+
+    const onFormSubmit = (data) => {
+        const ingredientAndMeasures = data.strIngredients.replaceAll(' ', '').split(',');
+
+        const splittedIngredients = [];
+        ingredientAndMeasures.forEach(item => {
+            splittedIngredients.push(item.split('-'))
+        })
+
+        const ingredients = ingredientIndexes.map((ingIndex) => {
+            return { [`strIngredient${ingIndex}`]: splittedIngredients[ingIndex - 1] ? splittedIngredients[ingIndex - 1][0] : '' }
+        })
+        const measures = ingredientIndexes.map((ingIndex) => {
+            return { [`strMeasure${ingIndex}`]: splittedIngredients[ingIndex - 1] ? splittedIngredients[ingIndex - 1][1] : '' }
+        })
+
+        ingredients.forEach(ing => {
+            Object.assign(data, ing);
+        })
+
+        measures.forEach(measure => {
+            Object.assign(data, measure);
+        });
+
+        delete data.strIngredients;
+        console.log({ ...data, idMeal: createId(), strMealThumb: data.strMealThumb[0].name })
```

Comment:
Add `reset();` to empty the form after submission. You get it from useForm:
```
const {reset} = useForm();
```

Reply:
Actually here, I wanted to delete just strIngredients that we took in the first place. We modified and added to the data with different version like 'strIngredient1, strIngredient2 etc... It seemed unnecessary to keep that strIngredient there. What do you think?

Reply:
That's right but you also need to reset the entire form after it's submitted and the data is processed, otherwise the user will need to clean the form manually to add another recipe

Reply:
Now I understand :)